apps/omg_watcher_rpc/lib/web/validators/block_constraints.ex

```diff
@@ -0,0 +1,40 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.WatcherRPC.Web.Validator.BlockConstraints do
+  @moduledoc """
+  Validates `/block.all` query parameters
+  """
+
+  import OMG.Utils.HttpRPC.Validator.Base
+
+  @doc """
+  Validates possible query constraints, stops on first error.
+  """
+  @spec parse(%{binary() => any()}) :: {:ok, Keyword.t()} | {:error, any()}
+  def parse(params) do
+    constraints = [
```

Comment:
This file and this file: https://github.com/omisego/elixir-omg/blob/master/apps/omg_watcher_rpc/lib/web/validators/constraints.ex are starting to get weird together.

We're all aware of this, we need to clean it up, would be nice to add a TODO comment

Reply:
TODO won't make the clean up happen.

Reply:
true, creating an issue for this is probably better. thanks.

Reply:
@unnawut And so a TODO becomes and issue: https://github.com/omisego/elixir-omg/issues/1215